chapters/03-talking_to_servers/readme.md

```diff
@@ -0,0 +1,678 @@
+
+Finally we can do the show view:
+
+```html
+<h1>The solutions!</h1>
+
+<h2>1. <%= @solution.first %></h2>
+<h2>2. <%= @solution.last %></h2>
+```
+
+![Polynomial submission](./images/new_polynomial_5.png)
+
+It only prints out the first and the second solution to the problem. Nothing
+to write home about. But with this we have a fully functioning polynomial
+solution finder written completely by hand! This was a learning experience, but
+we still have open questions! Why did Rails prefix the input's names with `post`
+and put the real name in brackets?
+
+Ha! Rails does something very cool here, it builds a sub-hash. A form can
+contain many different inputs for many different contexts, not to mix them up
+we can put them in sub-hashes. E.g.:
+
+```html
+<form action="/buildings" method="POST">
+  <p>
+    <label for="manager[name]">Manager name:</label>
+    <input type="text" name="manager[name]" \>
+  </p>
+  <p>
+    <label for="manager[phone]">Manager phone:</label>
+    <input type="text" name="manager[phone]" \>
+  </p>
+  <p>
+    <label for="building[address]">Building address</label>
+    <input type="text" name="building[address]" \>
+  </p>
+  <p>
+    <label for="building[name]">Building name</label>
+    <input type="text" name="building[name]" \>
+  </p>
+
+  <input name="authenticity_token" type="hidden" value="<%= form_authenticity_token %>" \>
+
+  <button type="submit">Save!</button>
+</form>
+```
+
+Would give us the following `params` object:
+
+```ruby
+{
+  manager: {
+    name: "John Doe",
+    phone: "+112345678"
+  },
+  building: {
+    address: "1 Somewhere St., 00000 Somewheresville, Somewhere",
+    name: "ACME Corp Headquaters"
+  }
+}
+```
+
+We can also store arrays in a similar manner. By using empty brackets `[]` we
+tell Rails that we want duplicates to be handled like members of an `Array`.
+Otherwise if multiple fields with the same name are sent to the server only the
+last will be used.
+
+```html
+<form action="/competitions" method="POST">
+  <p>
+    <label for="participant[name][]">1st place:</label>
+    <input type="text" name="participant[name][]" \>
+  </p>
+  <p>
+    <label for="participant[name][]">2nd place:</label>
+    <input type="text" name="participant[name][]" \>
+  </p>
+  <p>
+    <label for="participant[name][]">3rd place:</label>
+    <input type="text" name="participant[name][]" \>
+  </p>
+
+  <input name="authenticity_token" type="hidden" value="<%= form_authenticity_token %>" \>
+
+  <button type="submit">Save!</button>
+</form>
+```
+
+Gives us the following `params` object:
+
+```ruby
+{
+  participant: {
+    name: [
+      "Alice",
+      "Bob",
+      "Clay"
+    ]
+  }
+}
+```
+
+Arrays in params are most useful when working with multiple images, say in an
+application that has a gallery. Instead of forcing the user to upload an image
+one-by-one, we can allow the user to select all the images they want and upload
+them all at once.
```

Comment:
I believe that example and explanation are quite abstract here. Parts of this explanation can be incorporated later, when using each loop in polynomial form.

Reply:
I can't think of another way to jump the topic. Any suggestion?

Reply:
Not really, as the loop is just a show of how to use ruby with rendering and forms. This can be explained when we cover image uploads. I wanted to go into this here, but I couldn't think of a use-case without utilizing relations, migrations and the databse. So I skipped it.

Reply:
I still believe that this is a bit offsetting and could be reorganized a bit to achieve better clarity.
As I'm also in a short supply of time and can't write the solution, let's just roll with this :)